ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx

```diff
@@ -74,15 +74,14 @@ export function NotificationsSettingsAllowNotifications({
   }, [isMetamaskNotificationsEnabled]);
 
   useEffect(() => {
-    if (isMetamaskNotificationsEnabled && !error) {
+    if (!error) {
       listNotifications();
     }
   }, [isMetamaskNotificationsEnabled, error, listNotifications]);
 
   const toggleNotifications = useCallback(async () => {
```

Comment:
Drive-by observation: This looks like it could be racey as it's not idempotent. Refactoring it from `toggleNotifications()` to `setNotificationsEnabled(boolean)` should make it more robust.